### PR TITLE
Introduce push_finished signal

### PR DESCRIPTION
**Pull request by MarcelGeo**
Introducing new push finished signal to handle successful pushes to projects.

### CODE REVIEW

File: server/mergin/tests/utils.py [hunk 1]
```diff
@@ -241,7 +243,14 @@ def upload_file_to_project(project, filename, client):
...
             url, data=f_data, headers={"Content-Type": "application/octet-stream"}
         )
-    assert client.post(f"/v1/project/push/finish/{upload_id}").status_code == 200
+    return (resp, client.post(f"/v1/project/push/finish/{upload_id}"))
+
+
+def upload_file_to_project(project, filename, client):
+    """Add test file to project - start, upload and finish push process"""
+    resp_push_start, resp_push_finish = push_file(project, filename, client)
+    assert resp_push_finish.status_code == 200
```
[the remark this answers is not in the document]
Comment (MarcelGeo) [reply server/mergin/tests/utils.py:252]: I am returning tuple from push to proper handling of responses for push start and finish. In some tests I need chcem also response from push. I don't want to push manually.

[the remark this answers is not in the document]
Comment (MarcelGeo) [reply server/mergin/tests/utils.py:252]: This method is used for cases, where you do not need to handle response object.

File: server/mergin/tests/utils.py [hunk 2]
```diff
@@ -226,13 +224,29 @@ def upload_file_to_project(project, filename, client):
         "version": ProjectVersion.to_v_name(project.latest_version),
         "changes": changes,
     }
+    return data
+
+
+def push_file_start(project, filename, client, mocked_changes_data=None) -> Tuple:
```
Comment (varmar05) [inline server/mergin/tests/utils.py:230]: add docstring

File: server/mergin/tests/utils.py [hunk 3]
```diff
@@ -226,13 +224,29 @@ def upload_file_to_project(project, filename, client):
...
+    return resp
+
+
+def upload_file_to_project(project, filename, client):
+    """Add test file to project - start, upload and finish push process"""
+    file = os.path.join(test_project_dir, filename)
+    data = mock_changes_data(project, filename)
+    changes = data.get("changes")
+    resp = push_file_start(project, filename, client, data)
+    print(resp.json)
```
Comment (varmar05) [inline server/mergin/tests/utils.py:248]: rm

File: server/mergin/tests/utils.py [hunk 4]
```diff
@@ -241,7 +255,9 @@ def upload_file_to_project(project, filename, client):
         client.post(
             url, data=f_data, headers={"Content-Type": "application/octet-stream"}
         )
-    assert client.post(f"/v1/project/push/finish/{upload_id}").status_code == 200
+    push_finish = client.post(f"/v1/project/push/finish/{upload_id}")
+    assert resp.status_code == 200
+    assert push_finish.status_code == 200
```
Comment (varmar05) [inline server/mergin/tests/utils.py:260]: let's also return resp instead, it might comes in handy

**Review by varmar05 — CHANGES_REQUESTED** [review]

**Review by varmar05 — APPROVED** [review]

**Pull request merged**